.github/workflows/changesets-pr.yaml

```diff
@@ -0,0 +1,141 @@
+name: "Changesets PR"
+
+on:
+  push:
+    branches:
+      - master
+      - release/*
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  changesets-pr:
+    runs-on: ubuntu-latest
+    permissions: write-all
+    # First check ensures that the workflow runs only if the commit is NOT the changesets PR commit.
+    # Second check ensures that the workflow runs only after a commit is pushed into the branch,
+    # and not when the branch is created.
+    # This is to avoid running the workflow when a release/* branch is created.
+    if: |
+      github.event.head_commit.message != 'ci(release): versioning packages and changesets' &&
+      github.event.before != '0000000000000000000000000000000000000000'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: CI Setup
+        uses: ./.github/actions/ci-setup
+
+      # if there are changesets present, package.json will be bumped
+      - name: Bump and Collect Versions
+        run: |
+          pnpm changeset version
+          echo "RELEASE_VERSION=v$(sed -nE 's/^\s*"version": "(.*?)",$/\1/p' packages/fuels/package.json)" >> $GITHUB_ENV
+          echo "FUEL_CORE_VERSION=$(cat ./internal/fuel-core/VERSION)" >> $GITHUB_ENV
+          echo "FORC_VERSION=$(cat ./internal/forc/VERSION)" >> $GITHUB_ENV
+          git reset --hard
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set branch name and changeset PR title
+        run: |
+          echo "CHANGESET_PR_TITLE=$(echo "ci(release): \`${{ env.RELEASE_VERSION }}\` @ \`${{ github.ref_name }}\`")" >> $GITHUB_ENV
+
+      - name: Build
+        run: pnpm build
+
+      - name: Create Release Pull Request
+        id: changesets
+        uses: FuelLabs/changesets-action@main
+        with:
+          publish: pnpm changeset:publish ${{ env.RELEASE_VERSION }} ${{ github.ref_name }}
+          version: pnpm changeset:version-with-docs
+          commit: "ci(release): versioning packages and changesets"
+          title: ${{ env.CHANGESET_PR_TITLE }}
+          createGithubReleases: aggregate
+          githubReleaseName: ${{ env.RELEASE_VERSION }}
+          githubTagName: ${{ env.RELEASE_VERSION }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_DEPLOY_TOKEN }}
+
+      - name: Prettify changelog
+        run: pnpm changeset:update-changelog
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REF_NAME: ${{ github.ref_name }}
+          PUBLISHED: "false"
```

Comment:
I will be splitting the `changeset:update-changelog` command into two in a subsequent PR, one command for updating the changesets PR's changelog and another for updating the GitHub release's changelog. The current command does both based on this `PUBLISHED` flag, but we don't need this flag anymore as the workflow has been split.